### PR TITLE
Added Checkov

### PR DESCRIPTION
**Pull request by david-sandy**
This is the integration of the terraform scanner into this pipeline.
There should not be any terraform code here, but, in case that changes.
This is part of a task I was assigned to add Checkov to scan terraform code in all repositories that use buildkite.

### Checklist

* [ ] [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) updated
* [ ] [K8s Upgrade notes updated](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md)
* [ ] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
* [ ] All images have a valid tag and SHA256 sum

### CODE REVIEW

File: .buildkite/ci-checkov.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# In case it reports already installed
+asdf shell python 3.10.0
+
+# Install and run the plugin for checkov
+# Use the full path to run pip3.10
+/root/.asdf/installs/python/3.10.0/bin/pip3.10 install checkov
```
Comment (bobheadxi) [inline .buildkite/ci-checkov.sh:8]: Please see @jhchabran 's comment here: https://github.com/sourcegraph/sourcegraph/pull/30059#discussion_r791130925

File: .buildkite/ci-checkov.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# --quiet: Only show failing tests
+# --compact: Do not show code snippets
+# --framework: Only scan terraform code
+
+# Capture the error code
+CHECKOV_EXIT_CODE="$?"
+
+# We check the exit code and display a warning if anything was found
+if [[ "$CHECKOV_EXIT_CODE" != 0 ]]; then
+  buildkite-agent annotate 'Possible Terraform security issues found.  Please refer to the Sourcegraph handbook for guidance <a target="_blank" href="https://handbook.sourcegraph.com/product-engineering/engineering/cloud/security/checkov">here</a>.' --style 'warning' --context 'ctx-warn'
```
Comment (bobheadxi) [inline .buildkite/ci-checkov.sh:32]: same comment as https://github.com/sourcegraph/sourcegraph/pull/30059#discussion_r791124476

File: .buildkite/ci-checkov.sh (hunk 1 above)
Comment (david-sandy) [reply .buildkite/ci-checkov.sh]: Updated.

File: .buildkite/ci-checkov.sh (hunk 2 above)
Comment (david-sandy) [reply .buildkite/ci-checkov.sh]: Updated

File: .buildkite/pipeline.yaml [hunk 3]
```diff
@@ -27,6 +27,12 @@ steps:
     command: .buildkite/check-image-names.sh
     agents: { queue: standard }
 
+   # This runs the Checkov Terraform Code scanner
+  # https://www.checkov.io/
+  - command: .buildkite/ci-checkov.sh
+    label: ':lock: security - checkov' 
+    agents: { queue: "standard" }
```
Comment (bobheadxi) [inline .buildkite/pipeline.yaml:34]:
```suggestion
    agents: { queue: "standard" }
```

**Review by bobheadxi — APPROVED** [review]
nice :) minor formatting nit but otherwise lgtm!

**Pull request merged**